pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -351,7 +351,7 @@ func (m *ManagerImpl) Start(activePods ActivePodsFunc, sourcesReady config.Sourc
 	// Loads in allocatedDevices information from disk.
 	err := m.readCheckpoint()
 	if err != nil {
-		klog.InfoS("Continue after failing to read checkpoint file. Device allocation info may NOT be up-to-date", "err", err)
+		klog.ErrorS(err, "Continue after failing to read checkpoint file. Device allocation info may NOT be up-to-date")
```

Comment:
the nuance we lose here is the different log level in the logs, these will be now presented as ERRs which may be caught differently from alerts/monitoring tools. OTOH all this recoverable errors ARE worth an alert at least (we saw similar things in other resource managers).

Arguably, we should actually take the error path and do not swallow these errors, but this will change a long established behavior, so the fix can be worse than the issue.

TL;DR: I'm happier with this change than with the existing code.

Reply:
I agree that changing the log level to `ERR` might impact monitoring systems differently, but given that these recoverable errors are still significant, I think they warrant better visibility. Other resource managers already follow a similar pattern, so this aligns with existing behavior elsewhere.

That said, I understand the concern about modifying long-standing behavior. I think we should capture this in the resource management logging guideline that I would like to publish soon. 

Let me know if you think there's a better way to handle this or if adding to the guideline addresses your concerns.

Reply:
Clarification: the current change I think is beneficial for the reasons above. We may need a targeted revert of this very commit in the worst case, which is bearable.

The change of long-established behavior would be a followup hypotethical change on which we actually err out in the error paths on which currently we log ERR and continue. THAT would change a long established (debatable) behavior and THAT fix could likely be worse than the small "issue" this current change here is provoking.

So I think we should go along with this change with no further action until (if ever) further evidence comes out